File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        if(height.length<2 || height.length>100000) return 0;
        int N = 0, max = 0;
        for(int i=0; i<height.length; i++){
            for(int j=height.length-1; j>i; j--){
                if(height[i]<=height[j]){
                    N = (j-i) * height[i];
                    if(N>max) max = N;
                    break;
                }
            }
        }
        for(int j=height.length-1; j>=0; j--){
            for(int i=0; i<j; i++){
                if(height[j]<=height[i]){
                    N = (j-i) * height[j];
                    if(N>max) max = N;
                    break;
                }
                
            }
        }
        return max;
    }
}

File: 11-container-with-most-water/11-container-with-most-water.c


int maxArea(int* height, int heightSize){
    int *head = height, *tail = height + heightSize - 1;
    int max = 0, water = 0;
    while(head != tail){
        int len = tail - head;
        water = *head < *tail ? (*head * len) : (*tail * len);
        if(water > max) max = water;
        if(*head < *tail){
            head = head + 1;
        }else{
            tail = tail - 1;
        }
    }
    return max;
}